crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
     }
 }
 
-/// Parse and run command line arguments
-pub fn run() -> sc_cli::Result<()> {
-    let mut cli = Cli::from_args();
-
+fn set_madara_path(cli: &mut Cli) {
     // alias madara_path <> base_path
-    // TODO also alias tmp (tmp generates random base_paths that are not specified within
-    // the command)
-    let madara_path = match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
+    match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
         (Some(madara_path), _) => {
-            cli.run.run_cmd.shared_params.base_path = Some(madara_path.clone());
-            madara_path.to_str().unwrap().to_string()
+            cli.run.run_cmd.shared_params.base_path = Some(madara_path);
         }
         (_, Some(base_path)) => {
-            cli.run.madara_path = Some(base_path.clone());
-            base_path.to_str().unwrap().to_string()
+            cli.run.madara_path = Some(base_path);
         }
         _ => {
             let home_path = std::env::var("HOME").unwrap_or(std::env::var("USERPROFILE").unwrap_or(".".into()));
             let path = format!("{}/.madara", home_path);
             cli.run.run_cmd.shared_params.base_path = Some((path.clone()).into());
-            cli.run.madara_path = Some((path.clone()).into());
-            path
+            cli.run.madara_path = Some(path.into());
         }
     };
+}
+
+fn set_dev_environment(cli: &mut Cli) {
+    // create a reproducible dev environment
+    cli.run.run_cmd.shared_params.dev = false;
+    cli.run.run_cmd.shared_params.chain = Some("dev".to_string());
+
+    cli.run.run_cmd.force_authoring = true;
+    cli.run.run_cmd.alice = true;
+
+    // we can't set `--rpc-cors=all`, so it needs to be set manually if we want to connect with external
```

Comment:
How comes?

Reply:
That is not new. You need to set an Cors enum that doesn't have public visibility